src/App.tsx

```diff
@@ -17,18 +46,35 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                currentStatus={filterStatus}
+                onFilterChange={setFilterStatus}
+                query={query}
+                onQueryChange={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!isLoading && todos.length > 0 ? (
+                <TodoList
+                  todos={getFilteredTodos()}
+                  selectedTodo={isSelectedTodo}
+                  setSelectedTodo={setIsSelectedTodo}
+                />
+              ) : (
+                <Loader />
```

Comment:
The `Loader` component is used here to indicate loading state. Ensure that this component behaves as expected and provides a clear indication to users when data is being loaded. If there are any issues with its behavior, consider reviewing its implementation.